include/dali/c_api.h

```diff
@@ -135,7 +135,13 @@ enum {
   /**
    * If provided CPU memory is page-locked
    */
-  DALI_ext_pinned = (1<<1)
+  DALI_ext_pinned = (1<<1),
+
+  /**
+   * If provided, a CUDA copy kernel will be used to feed external source instead of cudaMemcpyAsync
+   * Only relevant when the input is either pinned host memory or device memory
+   */
+  DALI_use_copy_kernel = (1<<2),
```

Comment:
> It's hard for me to imagine a scenario, when user would prefer to use cudaMemcpy approach instead of ScatterGather, when having input in the Pinned memory

If you already occupied with compute DMA could ba faster. How about Xavier or a GPU with smaller number os SMs?

Reply:
I'd say that most of the time you prefer cudaMemcpy over scatter-gather kernel. The GPU might be doing expensive computations and have the DMA free to use. That should be the default case.